python/lsst/faro/measurement/DetectorTableMeasurement.py

```diff
@@ -0,0 +1,144 @@
+    """Configuration for DetectorTableMeasurementTask."""
+
+    columns = pexConfig.ListField(doc="Columns from sourceTable_visit to load.",
+                                  dtype=str, default=['coord_ra', 'coord_dec', 'detector'])
+
+    refObjLoader = pexConfig.ConfigurableField(
+        target=LoadReferenceCatalogTask,
+        doc="Reference object loader",
+    )
+
+    def setDefaults(self):
+        self.refObjLoader.refObjLoader.ref_dataset_name = "gaia_dr2_20200414"
+        self.refObjLoader.refObjLoader.requireProperMotion = True
+        self.refObjLoader.refObjLoader.anyFilterMapsToThis = 'phot_g_mean'
+        self.refObjLoader.doApplyColorTerms = False
+
+    def validate(self):
+        super().validate()
+        if 'detector' not in self.columns:
+            msg = "The column `detector` must be appear in the list of columns."
+            raise pexConfig.FieldValidationError(DetectorTableMeasurementConfig.columns, self, msg)
+        if self.connections.refDataset != self.refObjLoader.refObjLoader.ref_dataset_name:
+            msg = "The reference datasets specified in connections and reference object loader must match."
+            raise pexConfig.FieldValidationError(DetectorTableMeasurementConfig.columns, self, msg)
+
+
+class DetectorTableMeasurementTask(CatalogMeasurementBaseTask):
+    """Base class for science performance metrics measured on single-dector source catalogs."""
+
+    ConfigClass = DetectorTableMeasurementConfig
+    _DefaultName = "detectorTableMeasurementTask"
+
+    def run(self, catalog, refcat, refcatCalib):
+        return self.measure.run(self.config.connections.metric, catalog,
+                                refcat=refcat, refcatCalib=refcatCalib)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        catalog = inputs['catalog'].get(parameters={'columns': self.config.columns})
+        selection = (catalog['detector'] == butlerQC.quantum.dataId['detector'])
+        catalog = catalog[selection]
+
+        if self.config.connections.refDataset != '':
+            center = lsst.geom.SpherePoint(butlerQC.quantum.dataId.region.getBoundingCircle().getCenter())
+            radius = butlerQC.quantum.dataId.region.getBoundingCircle().getOpeningAngle()
+
+            epoch = butlerQC.quantum.dataId.records['visit'].timespan.begin
+
+            loaderTask = LoadReferenceCatalogTask(
+                config=self.config.refObjLoader,
+                dataIds=[ref.datasetRef.dataId
+                         for ref in inputRefs.refcat],
+                refCats=inputs.pop('refcat'))
+
+            # Catalog with proper motion and color terms applied
+            refcatCalib = loaderTask.getSkyCircleCatalog(
+                center,
+                radius,
+                [butlerQC.quantum.dataId.records['physical_filter'].name],
+                epoch=epoch)
+
+            # Next get unformatted catalog w/ all columns
```

Comment:
Why do you need both `refcat` and `refcatCalib`? How are they to be used differently, since they nominally contain the same data?

Reply:
The latter object does NOT include all of the columns from the original reference catalog, but rather only the coordinates and colorterm corrected magnitudes. If we want any of the other columns, we need both versions of the catalog.